### ISSUE TITLE
Theme inheritance for custom types seems broken

### ISSUE THREAD

**Issue by dotlogix** [godotengine/godot#67010]
### Godot version

4.0.dev (17c62a692)

### System information

Windows 10, Vulkan, AMD Radeon RX 6700 XT

### Issue description

**TLDR:**
In the inspector and theme editor custom definitions for inherited classes as well as variants are not considered correctly.
They don't show up in the inspectors "Theme Overrides"
Partially in the theme editor with ("Show Defaults")
Fully in the import tab of the theme editor

**Story:**
Atm I work on a rather complex control hierachy with several levels of inheritance and theme overrides at some places. However there is a strange thing going on there...

To get the editor working with custom defined theme properties at all, I created a singleton gdscript to register the types and variants using code.

**BaseTypes:**
For base types defined in cpp it is pretty easy. I just need to call the appropriate method in the setup script to add my own properties and that works pretty consistently.
They show up in the inspector under "Theme Overrides" as well as in the theme editor when "ShowDefault" is enabled.

**Script Class:**
For inherited types with a "class_name" it starts to behave in a weird unpredictable way.
You can define a theme type "Script" and add some additional values.

In the inspector it only shows theme items added to the underlying cpp class (Control), but not the one from the scripts class_name (ScriptClass).

In the theme editor it shows the one of the class_name (ScriptClass) but not from the base cpp class (Control).
If you however set the base type of "ScriptClass" to the cpp class explicitly this turns around and it stops showing the class_name (ScriptClass) ones but the base type (Control) ones appear.

**Inherited Class**
In the theme editor it shows the one of the class_name (InheritedClass) but not from the base cpp class (ScriptClass).
If you however set the base type of "InheritedClass" to the "ScriptClass" explicitly this turns around and it stops showing the class_name (InheritedClass) ones but the base type (ScriptClass) ones appear.

**Variants:**
Basically the same as for **Inherited Class**

**Other**
The project theme configuration, manually applied themes, custom types as well as type variants are also not considered. Added definitions neither affect the "Theme Overrides" nor the "Show Default" option in the theme editor.

I guess the underlying issue has sth todo how Control::_get_property_list is setup atm
more. I think it would make much more sense if all properties are shown or the full hierarchy Variants and Types for the project theme, default theme as well as manually attached themes and custom definitions in the theme editor itself.

I think "get_theme_icon" already does that to some extent, but not sure.

### Steps to reproduce

I tried my best to set this project up as good as I can, buggy.theme basically shows the issue with the theme editor. I just added the overrides so the types are persisted. Empty definitions get removed on save somehow.

### Minimal reproduction project

[ThemeInheritance.zip](https://github.com/godotengine/godot/files/9729418/ThemeInheritance.zip)

**Comment by linkpy** [comment]
I dont know if it's related but i'm having an issue at run-time with custom theme type inheritance : 
![image](https://user-images.githubusercontent.com/2700596/210294825-e03efcd5-53ff-41aa-bc3a-ad07dc8e8b01.png)

The type `Tile` has a stylebox `regular`, and the type `Token` inherits it, but trying to access it with `get_theme_stylebox("regular", "Token")` doesn't work but after looking for how theme resources and inheritance works i believe it should.

**Comment by WhalesState** [comment]
creating a custom theme type inside a theme has no effect too!

**Comment by Calinou** [comment]
> creating a custom theme type inside a theme has no effect too!

Please continue the discussion in https://github.com/godotengine/godot/issues/78308.

**Comment by Jan-PieterInghels** [comment]
I can confirm this is still an issue in 4.2.2.
### Works

- Creating a new type and setting the base type to an engine native type.

### Does not work

- Creating a new type and setting the base type to a custom type.



Not working means that no 'items' (eg. font_color) appear in any of the items tabs or the manage theme items menu, as if nothing is inherited from the base type.

I haven not found a workaround so far, so only using engine native types as the base type seems to be the only solution for now.

**Comment by WinnerWind** [comment]
Hi, just struggled with this issue, and I'm not sure what's causing it. 
Using Engine Native types is a solution, but if that's the case, the docs need to be amended.
> Type variations can also chain and extend other type variations. 
[here](https://docs.godotengine.org/en/stable/tutorials/ui/gui_theme_type_variations.html)

**Comment by The-Stubbs** [comment]
> I can confirm this is still an issue in 4.1.1 stable and 4.2.2 stable.

This is still the case in 4.5.dev4
